File: PYTHON/06-POO/encapsulamiento/persona.py
class Persona:

    # Los atributos Privados se definen con el signo __, ya que en el lenguaje no existe una palabra
    # reservada para tal fin.
    __nombre = None
    __Edad = None

    def __init__(self, nombre, edad):
        self.__nombre = nombre
        self.__edad = edad

    def __metodo_privado(self):
        print('Soy un método privado')

# Para poder acceder y/o modificar los atributos privados de la clase, debemos
# crear método públicos
    def get_nombre(self):
        return self.__nombre

    def set_nombre(self, nombre):
        self.__nombre = nombre

    def get_edad(self):
        return self.edad

    def set_edad(self, edad):
        self.edad = edad



    def __str__(self):
        return f'Nombre: {self.__nombre} \nEdad: {self.__edad}'

File: GO/src/golang/07-modularizacion/models/persona.go
package models

type Persona struct {
	nombre string
	edad   int
}

func (p *Persona) Constructor(nombre string, edad int) {
	p.nombre = nombre
	p.edad = edad
}

// Obtener el valor de un atributo mediante una función GET
func (p *Persona) GetNombre() string {
	return p.nombre
}

// Modificar el valor de un atributo mediante una función SET
func (p *Persona) SetNombre(nombre string) {
	p.nombre = nombre
}

// Obtener el valor de un atributo mediante una función GET
func (p *Persona) GetEdad() int {
	return p.edad
}

// Modificar el valor de un atributo mediante una función SET
func (p *Persona) SetEdad(edad int) {
	p.edad = edad
}
